core/unit_test/tools/TestProfilingSection.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <Kokkos_Profiling_ProfileSection.hpp>
+
+#include <gtest/gtest.h>
+
+namespace {
+struct Section {
+  std::string name;
+  int start_call_cnt;
+  int stop_call_cnt;
+  int is_destroyed;
+  friend std::ostream& operator<<(std::ostream& os, Section const& s) {
+    os << "( " << s.name << ", " << s.start_call_cnt << ", " << s.stop_call_cnt
+       << ", " << s.is_destroyed << " )";
+    return os;
+  }
+  friend bool operator==(Section const& l, Section const& r) {
+    return (l.name == r.name) && (l.start_call_cnt == r.start_call_cnt) &&
+           (l.stop_call_cnt == r.stop_call_cnt) &&
+           (l.is_destroyed == r.is_destroyed);
+  }
+};
```

Comment:
@masterleinad I changed this because tuple comparison in google test wasn't great
```
../core/unit_test/tools/TestProfilingSection.cpp:114: Failure
Value of: Section("two", 1, 0, 2)
  Actual: 48-byte object <02-00 00-00 00-00 00-00 01-00 00-00 00-00 00-00 B0-98 FF-FF FF-7F 00-00 03-00 00-00 00-00 00-00 74-77 6F-00 00-00 00-00 18-22 55-00 00-00 00-00>
Expected: kokkosp_test_section_vector[1]
Which is: 48-byte object <01-00 00-00 00-00 00-00 01-00 00-00 00-00 00-00 20-FE 55-00 00-00 00-00 03-00 00-00 00-00 00-00 74-77 6F-00 00-00 00-00 00-00 00-00 00-00 00-00>
```
VS now
```
./core/unit_test/tools/TestProfilingSection.cpp:128: Failure
Value of: (Section{"two", 1, 1, 1})
  Actual: ( two, 1, 1, 1 )
Expected: kokkosp_test_section_vector[1]
Which is: ( two, 1, 0, 1 )